src/acpi.rs

```diff
@@ -419,3 +424,357 @@ pub fn process_msct() -> (
+            self.data1,
+            self.data2,
+            self.data3,
+            self.index8,
+            self.index9,
+            self.index10,
+            self.index11,
+            self.index12,
+            self.index13,
+            self.index14,
+            self.index15
+        )
+    }
+}
+
+fn from_slice_u8_to_Guid(slice: &[u8]) -> Guid {
+    let p: *const [u8; core::mem::size_of::<Guid>()] =
+        slice.as_ptr() as *const [u8; core::mem::size_of::<Guid>()];
+    unsafe { core::mem::transmute(*p) }
+}
+
+// To parse DeviceHandle in ACPI_NFIT_MEMORY_MAP and ACPI_NFIT_FLUSH_ADDRESS subtables
+const ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK: u32 = 0xf;
+const ACPI_NFIT_DEVICE_HANDLE_12BIT_MASK: u32 = 0xfff;
+
+// NFIT subtable type = 0x0
+fn parse_nfit_spa_range_structure(entry: *const ACPI_NFIT_SYSTEM_ADDRESS) -> MemoryDescriptor {
+    let mut mem_desc = MemoryDescriptor::default();
+    unsafe {
+        assert_eq!((*entry).Header.Type, 0);
+        debug!(
+            "ACPI_NFIT_SYSTEM_ADDRESS - Type {}, Length {},
+                RangeIndex: {},
+                Flags: {},
+                ProximityDomain: {},
+                RangeGuid: {:x?},
+                Address: 0x{:x},
+                Length: {} Bytes,
+                MemoryMapping: {:?}",
+            (*entry).Header.Type,
+            (*entry).Header.Length,
+            (*entry).RangeIndex,
+            (*entry).Flags,
+            (*entry).ProximityDomain,
+            from_slice_u8_to_Guid(&(*entry).RangeGuid),
+            (*entry).Address,
+            (*entry).Length,
+            MappingAttributes::from((*entry).MemoryMapping)
+        );
+
+        assert_eq!(
+            (*entry).Length % BASE_PAGE_SIZE as u64,
+            0,
+            "Not multiple of page-size."
+        );
+        mem_desc.ty = MemoryType::PERSISTENT_MEMORY;
+        mem_desc.phys_start = (*entry).Address;
+        mem_desc.page_count = (*entry).Length / BASE_PAGE_SIZE as u64;
+        mem_desc.att = MemoryAttribute::from_bits_truncate((*entry).MemoryMapping);
+
+        mem_desc
+    }
+}
+
+// NFIT subtable type = 0x1
+fn parse_nfit_region_mapping_structure(entry: *const ACPI_NFIT_MEMORY_MAP) {
+    // To parse Flags
+    const ACPI_NFIT_MEM_SAVE_FAILED: u32 = 0x1; /* 00: Last SAVE to Memory Device failed */
+    const ACPI_NFIT_MEM_RESTORE_FAILED: u32 = 0x1 << 1; /* 01: Last RESTORE from Memory Device failed */
+    const ACPI_NFIT_MEM_FLUSH_FAILED: u32 = 0x1 << 2; /* 02: Platform flush failed */
+    const ACPI_NFIT_MEM_NOT_ARMED: u32 = 0x1 << 3; /* 03: Memory Device is not armed */
+    const ACPI_NFIT_MEM_HEALTH_OBSERVED: u32 = 0x1 << 4; /* 04: Memory Device observed SMART/health events */
+    const ACPI_NFIT_MEM_HEALTH_ENABLED: u32 = 0x1 << 5; /* 05: SMART/health events enabled */
+    const ACPI_NFIT_MEM_MAP_FAILED: u32 = 0x1 << 6; /* 06: Mapping to SPA failed */
+
+    unsafe {
+        assert_eq!((*entry).Header.Type, 1);
+        debug!(
```

Comment:
Can we just implement debug on type to put actual print logic there?

Reply:
Should be done in the acpica-sys repo?